### PR TITLE
Unify chat's and inline edit dialog's prompt history managers

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://linear.app/sourcegraph/issue/CODY-2129/inline-edit-dialog-up-and-down-arrows-does-not-work-properly-with-the.

## Test plan
1. Test prompt history in the chat and in the inline edit dialog

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/edit/InstructionsInputTextArea.kt [hunk 1]
```diff
@@ -68,14 +68,6 @@ class InstructionsInputTextArea(parentDisposable: Disposable) :
     }
   }
 
-  // This is used by the up/down arrow keys to insert a history item.
-  fun setTextAndSelectAll(newContents: String?) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/InstructionsInputTextArea.kt:72]: no longer needed

File: src/main/kotlin/com/sourcegraph/cody/edit/HistoryManager.kt [hunk 2]
```diff
@@ -1,49 +0,0 @@
-package com.sourcegraph.cody.edit
-
-/** Manages user prompt history in memory for a session. */
-class HistoryManager<T>(private val capacity: Int) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/HistoryManager.kt:4]: removed in favor of [TextAreaHistoryManager.kt](https://github.com/sourcegraph/jetbrains/pull/1786/files#diff-7ee152a571f6e3a8ba8a5f2cf1d1e9642da2ed4b8c1164601fef2ba01a020e74) and [ChatPromptHistory.kt](https://github.com/sourcegraph/jetbrains/pull/1786/files#diff-2a9606f41be2726cbcb7152452526de41192dbdd371f85a8c8f5fa30aad16f58)

File: src/main/kotlin/com/sourcegraph/cody/chat/CodyChatMessageHistory.kt [hunk 3]
```diff
@@ -1,85 +0,0 @@
-package com.sourcegraph.cody.chat
-
-import com.intellij.openapi.project.Project
-import com.intellij.ui.components.JBTextArea
-import com.sourcegraph.cody.history.HistoryService
-import com.sourcegraph.cody.history.state.MessageState
-import java.util.*
-
-class CodyChatMessageHistory(
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/CodyChatMessageHistory.kt:9]: removed in favor of [TextAreaHistoryManager.kt](https://github.com/sourcegraph/jetbrains/pull/1786/files#diff-7ee152a571f6e3a8ba8a5f2cf1d1e9642da2ed4b8c1164601fef2ba01a020e74) and [ChatPromptHistory.kt](https://github.com/sourcegraph/jetbrains/pull/1786/files#diff-2a9606f41be2726cbcb7152452526de41192dbdd371f85a8c8f5fa30aad16f58)

File: src/main/kotlin/com/sourcegraph/cody/edit/EditCommandPrompt.kt [hunk 4]
```diff
@@ -378,34 +375,14 @@ class EditCommandPrompt(
...
-                clearActivePrompt()
+                performCancelAction()
               }
             }
-            updateOkButtonState()
           }
         })
   }
 
-  override fun getRootPane(): JRootPane {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/EditCommandPrompt.kt:392]: this was redundant in the first place

File: src/main/kotlin/com/sourcegraph/cody/edit/EditCommandPrompt.kt [hunk 5]
```diff
@@ -193,32 +207,16 @@ class EditCommandPrompt(
...
-        }
-
-        private fun handleDocumentChange() {
-          runInEdt {
-            updateOkButtonState()
-            checkForInterruptions()
-          }
-        }
-      }
+  private val promptCaretListener = CaretListener {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/EditCommandPrompt.kt:210]: a simpler solution similar to the one in PromptPanel

File: src/test/kotlin/com/sourcegraph/cody/chat/PromptHistoryTest.kt
Comment (mkondratek) [inline src/test/kotlin/com/sourcegraph/cody/chat/PromptHistoryTest.kt:1]: I would be a code removal but I added tests 😅

File: src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  fun getPrevious(): String? {
+    if (currentIndex > 0) {
+      currentIndex--
+    }
+    return history.getOrNull(currentIndex)
+  }
+
+  fun getNext(): String? {
+    if (currentIndex < history.size) {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt:23]: Shouldn't it be `if (currentIndex < history.size - 1)`?

Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt:23]: No, we want to get `null` at the end here. Based on that we determine that we want to show a clear input field again.

e.g. with the history:
history[0] = "hi"
history[1] = "sup?"
and the text area:
textArea.text = ""

1. ⬆️ 
2. textArea.text = "sup?"
3. ⬇️ 
4. textArea.text = ""

Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt:23]: Ok, cool. It a was a bit misleading to me that `getPrevious` have the same signature and code layout, but in it's case we prevent the index to go bellow 0. 
But I understand the intention is that `null` can happen there if `history` is empty?

**Review by pkukielka — APPROVED** [review]
LGTM

File: src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt (hunk 6 above)
Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt:23]: Yes, it is possible but handled by the history manager and covered by tests (unit tests and my manual tests)

**Pull request merged**